.github/workflows/awaiting-user-response.yml

```diff
@@ -0,0 +1,117 @@
+
+              // Check if the "AwaitingUserResponse" label is present.
+              const labels = issue.labels.map(label => label.name);
+              if (labels.includes('AwaitingUserResponse')) {
+                // Remove the label.
+                await github.rest.issues.removeLabel({
+                  owner,
+                  repo,
+                  issue_number: number,
+                  name: 'AwaitingUserResponse'
+                });
+              }
+            }
+
+      - name: Handle Scheduled Cleanup
+        if: github.event_name == 'schedule'
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const { owner, repo } = context.repo;
+
+            // Fetch all open issues with the "AwaitingUserResponse" label.
+            const issues = await github.paginate(github.rest.issues.listForRepo, {
+              owner,
+              repo,
+              state: 'open',
+              labels: 'AwaitingUserResponse',
+              per_page: 100
+            });
+
+            const now = new Date();
+
+            for (const issue of issues) {
+              const timelineEvents = await github.paginate(github.rest.issues.listEventsForTimeline, {
+                owner,
+                repo,
+                issue_number: issue.number,
+                per_page: 100
+              });
+
+              // Find the latest labeled event for "AwaitingUserResponse".
+              const labeledEvents = timelineEvents.filter(event =>
+                event.event === 'labeled' && event.label.name === 'AwaitingUserResponse'
+              );
+
+              if (labeledEvents.length > 0) {
+                const latestLabelEvent = labeledEvents.reduce((latest, event) =>
+                  new Date(event.created_at) > new Date(latest.created_at) ? event : latest
+                );
+
+                const labelAppliedAt = new Date(latestLabelEvent.created_at);
+		// TODO(glider): use diffDays instead of diffHours once we finish testing.
+                // const diffDays = (now - labelAppliedAt) / (1000 * 60 * 60 * 24);
+                const diffHours = (now - labelAppliedAt) / (1000 * 60 * 60);
+
+                //if (diffDays > 90) {
+                if (diffHours > 2) {
+                  // Close the issue.
+                  await github.rest.issues.update({
+                    owner,
+                    repo,
+                    issue_number: issue.number,
+                    state: 'closed'
+                  });
+
+                  // Add a comment notifying the user.
+                  await github.rest.issues.createComment({
+                    owner,
+                    repo,
+                    issue_number: issue.number,
+                    body: 'This issue has been automatically closed as it has been marked "AwaitingUserResponse" for more than 90 days with no activity.'
```

Comment:
should we add some comment to warn that this is about to happen? or explicitly say "if you are still experiencing this, reopen the bug" or something.